dynet/cfsm-builder.cc

```diff
@@ -233,6 +233,15 @@ void ClassFactoredSoftmaxBuilder::serialize(Archive& ar, const unsigned int) {
   ar & p_rc2ws;
   ar & p_rcwbiases;
 }
+
+
+void ClassFactoredSoftmaxBuilder::initialize_expressions() {
```

Comment:
Hmm, there's a bit of a disconnect here between "initialize_expressions" and the for loop, which simply creates and destroys temporary variables. Is there a cleaner and more intuitive way to do this? At the very least this should be commented.